File: Leetcode/645. Set Mismatch/solution.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        dup = None
        missing = 1

        for num in nums:
            if nums[abs(num)-1] < 0:
                dup = abs(num)
            else:
                nums[abs(num)-1] *= -1

        for i in range(1, len(nums)):
            if(nums[i] > 0):
                missing = i + 1

        return [dup, missing]

File: Leetcode/645. Set Mismatch/solution.rs
impl Solution {
    pub fn find_error_nums(mut nums: Vec<i32>) -> Vec<i32> {
        let mut dup = 0;
        let mut missing = 1;
        let mut nums2 = nums.clone();
        
        for i in 0..nums2.len() {
            let num = nums2[i].abs() as usize - 1;
            if(nums2[num] < 0) {
                dup = nums2[i].abs();
            }
            else {
                nums2[num] *= -1;
            }
        }
        
        for i in 1..nums2.len() {
            if nums2[i] > 0 {
                missing = i as i32 + 1;
            }
        }
        
        return vec![dup, missing];
    }
}